compiler/rustc_middle/src/hir/place.rs

```diff
@@ -113,3 +113,50 @@ impl<'tcx> Place<'tcx> {
         if projection_index == 0 { self.base_ty } else { self.projections[projection_index - 1].ty }
     }
 }
+
+pub fn place_to_string_for_capture(place: &Place<'tcx>, base_name: String) -> String {
```

Comment:
I'd move this out of here to `compiler/rustc_middle/src/ty/mod.rs` and then have the signature be 

```rust 
pub fn place_to_string_for_capture<'tcx>(tcx: TyCtxt<'tcx>, place: &Place<'tcx>) 
```

and then compute the var_name (replace base_name) using `tcx.hir().name(var_hir_id)`

Where we get the `var_hir_id` from `place.base`

Reply:
done